src/components/Avatar/index.tsx

```diff
@@ -0,0 +1,29 @@
+import classnames from 'classnames';
+import React, { FC } from 'react';
+
+import styles from './Avatar.module.css';
+
+import { IAvatarProps } from './IAvatar.d';
+
+const Avatar: FC<IAvatarProps> = ({
+  variant = 'text',
+  text = 'ww',
+  size = 'sm',
+  imageURL,
+  className,
+}) => {
+  const avatarClasses = classnames(
+    styles.shared,
+    styles[`size-${size}`],
+    styles[`variant-${variant}-shared`],
+    styles[`variant-${variant}-${size}`],
+    className
+  );
+  return variant === 'text' ? (
+    <p className={avatarClasses}>{text}</p>
```

Comment:
the business logic for this component is not choosing what we want to render whether text or image,
we want this component to render the first two chars if the image URL request returned any error, and if not it will render the image, so we need to use a function / hook which takes the URL and returns `success`, `error` or `pending` status and based on that we will render the right element